### DISCUSSION TITLE
design some indepedant places to host all the custom configurations

### DISCUSSION THREAD

**Discussion by copdips** [mmistakes/minimal-mistakes#3940] [Ideas]
Hello,

I've been using this amazing blog engine since a while, thanks a lot for the work.
I would like to submit a wish to have a single (or some) independant place(s) to host all the custom configurations instead of in-place overwritting.

Yes, many of the configurations could already been done by `_config.yml`, but still a lot of visual effet configurations should be tweaked in different files out of `_config.yml`, for example in `.scss`, `_layouts`, `navigation.html`, etc.

What I wish is to have somewhere that's in addition to all the existing files in the upstream repo (upstream is https://github.com/mmistakes/minimal-mistakes), that will overwrites the default configurations when starting up the engine.
By this way, all the overwrites are easily identified. It's especially a pain for me to `git rebase` to get the latest version from the upstream, because I have many conflits to handle. And if all the overwrites are in some independant places, this git conflit won't occur.

BTW, sometimes, I wanted to shared with others the customized visual effet tweak that I've done, but it's quite difficult to search in the git history what I've done.

**Comment by iBug** [comment]
If you're forking this theme repo to build your own site, chances are you're doing it the wrong way.

You should leverage Jekyll's support for customization while using the theme via the "remote theme" or Ruby gem method. Then you can place your overrides inside your own repository, and you won't have to worry about merging upstream changes as you're never *overwriting* any file.

For CSS/SCSS stuff, it's better that you define your own CSS rules that override the theme's setting. You can also override some SCSS variables before `@import`-ing the theme's SCSS parts.

You can check out [my website repo](https://github.com/iBug/iBug-source) as an example. I overrode a few `_layouts` and `_includes`, and extensively modified CSS to suit my needs. Still I have yet to encounter a "merge conflict" issue, and I can easily share with others my customizations.

**Reply by copdips** [reply]
@iBug , thanks a lot for your helpful reply, I'll take a look.

**Reply by copdips** [reply]
If I understand well, with the remote theme or gem based theme, for e.g, suppose I need to change the value of the var [$x-large: 1280px !default;](https://github.com/mmistakes/minimal-mistakes/blob/8a67ce8e41ec850f2d7c373aa47739b2abfee6f1/_sass/minimal-mistakes/_variables.scss#L151) to `1520` which is located in `_sass/minimal-mistakes/_variables.scss`, so I just need to create the same file in my repo with the same folder structure, and add this line `$x-large: 1520px !default;` in my own file `_variables.scss` ? And finally, my owne file `_variables.scss` will only contain a single line ?

**Reply by iBug** [reply]
This method is for file-based overriding.

For SCSS which has its own "mixing" support, you can just override `assets/css/main.scss` and prepend/append your own rules. See my example [here](https://github.com/iBug/iBug-source/blob/master/assets/css/main.scss).

Note that file-based overrides effectively *replace* the content with your local copy, so if you want to do it that way, you should copy the file from the theme and change its content, not putting a new file with only one line in it.

**Reply by copdips** [reply]
gotcha, i will test it this weekend, many thx.
